### PR TITLE
fix: use same aws-crt-swift version in aws-sdk-swift as smithy-swift

### PR DESCRIPTION
**Pull request by ganeshnj**
## Issue \#

Fixes https://github.com/awslabs/aws-sdk-swift/issues/820

### Motivation

aws-crt-swift added to aws-sdk-swift and smithy-swift, hence we have to update in both packages. Additionally, SPM started throwing warning on double inclusion of the dependency which will be covered to error in future versions.

Also, aws-crt-swift used as main and specific version smithy-swift which can cause further confusion.

### Modifications

- aws-crt-swift version is same on both aws-sdk-swift and smithy-swift
- every target list their dependencies explicitly rather depending on the transitive deps.

### Result

no warnings in SPM

In CI following 
```swift
let useLocalDeps = ProcessInfo.processInfo.environment["AWS_SWIFT_SDK_USE_LOCAL_DEPS"] != nil
let useMainDeps = ProcessInfo.processInfo.environment["AWS_SWIFT_SDK_USE_MAIN_DEPS"] != nil

switch (useLocalDeps, useMainDeps) {
case (true, true):
    fatalError("Unable to determine which dependencies to use. Please only specify one of AWS_SWIFT_SDK_USE_LOCAL_DEPS or AWS_SWIFT_SDK_USE_MAIN_DEPS.")
case (true, false):
    package.dependencies += [
        .package(path: "../smithy-swift")
    ]
case (false, true):
    package.dependencies += [
        .package(url: "https://github.com/awslabs/smithy-swift", branch: "main")
    ]
case (false, false):
    package.dependencies += [
        .package(url: "https://github.com/awslabs/smithy-swift", .exact("0.10.0"))
    ]
}
```

translated to
```swift
 let useLocalDeps = ProcessInfo.processInfo.environment["AWS_SWIFT_SDK_USE_LOCAL_DEPS"] != nil
let useMainDeps = ProcessInfo.processInfo.environment["AWS_SWIFT_SDK_USE_MAIN_DEPS"] != nil

switch (useLocalDeps, useMainDeps) {
case (true, true):
    fatalError("Unable to determine which dependencies to use. Please only specify one of AWS_SWIFT_SDK_USE_LOCAL_DEPS or AWS_SWIFT_SDK_USE_MAIN_DEPS.")
case (true, false):
    package.dependencies += [
        .package(path: "/Users/runner/work/aws-sdk-swift/aws-sdk-swift/target/build/deps/smithy-swift")
    ]
case (false, true):
    package.dependencies += [
        .package(url: "https://github.com/awslabs/smithy-swift", branch: "main")
    ]
case (false, false):
    package.dependencies += [
        .package(url: "https://github.com/awslabs/smithy-swift", .exact("0.10.0"))
    ]
}
```

## Conventional Commits

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by epau — APPROVED** [review]

File: scripts/generatePackageSwift.swift [hunk 1]
```diff
@@ -37,13 +37,6 @@ struct PackageDeps: Codable {
     var clientRuntimePath: String?
 }
 
-// When AWS_SDK_RELEASE_IN_PROGRESS is set, the manifest will require
-// exact versions of smithy-swift and aws-crt-swift no matter what
-// branches or paths are set.
-let releaseInProgress: Bool = {
-    env["AWS_SDK_RELEASE_IN_PROGRESS"] != nil
```
Comment (jbelkins) [inline scripts/generatePackageSwift.swift:44]: If this env var is no longer being read from, then we should remove the place where it is set in our release workflow

File: scripts/setup_for_crt_builder.sh [hunk 2]
```diff
@@ -6,6 +6,10 @@ set -x
 # with the CRT builder script (see https://github.com/awslabs/aws-crt-builder).
 # For use only as a pre-build step when using the CRT builder script on CI.
 
+# Enables the use of local dependencies, which are required when building with
+# the CRT builder script.
+export AWS_SWIFT_SDK_USE_LOCAL_DEPS=1
```
Comment (jbelkins) [inline scripts/setup_for_crt_builder.sh:11]: Releases use the CRT builder as well.  If this is set for a release, then a release will use whatever reference CRT builder retrieves & installs by default, not the pinned versions in `packageDependencies.plist`

**Review by jbelkins — CHANGES_REQUESTED** [review]
Not sure this will work properly for release.

File: scripts/generatePackageSwift.swift (hunk 1 above)
Comment (ganeshnj) [reply scripts/generatePackageSwift.swift:44]: good call, removed.

File: scripts/setup_for_crt_builder.sh (hunk 2 above)
Comment (ganeshnj) [reply scripts/setup_for_crt_builder.sh:11]: not entirely,

what this switch do, is basically manipulate 
```
let useLocalDeps = ProcessInfo.processInfo.environment["AWS_SWIFT_SDK_USE_LOCAL_DEPS"] != nil
let useMainDeps = ProcessInfo.processInfo.environment["AWS_SWIFT_SDK_USE_MAIN_DEPS"] != nil

switch (useLocalDeps, useMainDeps) {
case (true, true):
    fatalError("Unable to determine which dependencies to use. Please only specify one of AWS_SWIFT_SDK_USE_LOCAL_DEPS or AWS_SWIFT_SDK_USE_MAIN_DEPS.")
case (true, false):
    package.dependencies += [
        .package(path: "../smithy-swift") <------- THIS
    ]
case (false, true):
    package.dependencies += [
        .package(url: "https://github.com/awslabs/smithy-swift", branch: "main")
    ]
case (false, false):
    package.dependencies += [
        .package(url: "https://github.com/awslabs/smithy-swift", .exact("0.10.0"))
    ]
}
```

change the path to the dir where repo is located in the runner.

This code goes out in released Package.swift and when both AWS_SWIFT_SDK_USE_LOCAL_DEPS and AWS_SWIFT_SDK_USE_MAIN_DEPS are unset, which we expect to be in customer apps, we will use the pinned version.

**Review by jbelkins — APPROVED** [review]

**Pull request merged**